services/backend/app/crud/crud_recipe.py

```diff
@@ -66,5 +73,119 @@ def execQuery(self, q: Query, *, skip: int, limit: int) -> List[Recipe]:
     def get_all(self, db: Session):
         return db.query(Recipe).all()
 
+    def create(self, db: Session, *, obj_in: RecipeCreate, owner_id: UUID) -> Recipe:
+
+        db_obj = Recipe(
+            title=obj_in.title,
+            cookingTime=obj_in.cookingTime,
+            difficulty=obj_in.difficulty,
+            calories=obj_in.calories,
+            portion=obj_in.portion,
+            rating=obj_in.rating,
+            owner_id=owner_id,
+        )
+        db.add(db_obj)
+        db.commit()
+        db.refresh(db_obj)
+        for stage in obj_in.stages:
+            db_stage_obj = Stage(name=stage.name, content=stage.content, recipe=db_obj)
+            db.add(db_stage_obj)
+            db.commit()
+            db.refresh(db_stage_obj)
+
+            for product in stage.products:
+                print(product.name)
+                db_product_stage_obj = ProductsInStages(
+                    product_id=product.product_id,
+                    stage_id=db_stage_obj.id,
+                    amount=product.amount,
+                    amount_unit=product.amount_unit,
+                )
+                db.add(db_product_stage_obj)
+                db.commit()
+                db.refresh(db_product_stage_obj)
+
+        for tag in obj_in.tags:
+            # TODO check if tags are good
```

Comment:
Tags are good, what do you mean?, also i would keep tags uniqe by name, do you agree?

Reply:
It was leftover ...